Release 1.21.0 (#441)
scrapinghub/spidermon: CHANGES.rst

Release notes
=============

1.21.0 (2024-04-18)

> - bug: Fix monitor ailing when operating offset-naive and offset-aware datestimes (`PR#439 <https://github.com/scrapinghub/spidermon/pull/439>`_)

Reviewer: ```suggestion
- bug: Fix monitor failing when operating offset-naive and offset-aware datestimes (`PR#439 <https://github.com/scrapinghub/spidermon/pull/439>`_)
```

- bug: Fix TypeError in ``PeriodicItemCountMonitor`` when item_scraped_count was not defined the first time it was checked (`PR#436 <https://github.com/scrapinghub/spidermon/pull/436>`_)
- bug: Fix ``calculate_field_coverage`` incorrect handling of field names ending with _items (`PR#437 <https://github.com/scrapinghub/spidermon/pull/437>`_)
- bug: Fix monitor failing when operating offset-naive and offset-aware datetimes (`PR#439 <https://github.com/scrapinghub/spidermon/pull/439>`_)

> - bug: Fix tests for minimum properties and items in Jsonschema validations (`PR#432 <https://github.com/scrapinghub/spidermon/pull/432>`_)

Reviewer: ```suggestion
- bug: Fix tests for minimum properties and items in JSON Schema validations (`PR#432 <https://github.com/scrapinghub/spidermon/pull/432>`_)
```

> - feature: Add ``SPIDERMON_DICT_FIELDS_COVERAGE_LEVELS`` setting to limit nested dict stats computation (`PR#433 <https://github.com/scrapinghub/spidermon/pull/433>`_)

Reviewer: ```suggestion
- feature: Add the ``SPIDERMON_DICT_FIELDS_COVERAGE_LEVELS`` setting to limit nested dict stats computation (`PR#433 <https://github.com/scrapinghub/spidermon/pull/433>`_)
```

- bug: Fix CI/CD pipelines not working due to changes on Scrapy (`PR#426 <https://github.com/scrapinghub/spidermon/pull/426>`_)

> - feature: Add ``SendSNSNotification`` to Spidermon Notifications (`PR#413 <https://github.com/scrapinghub/spidermon/pull/413>`_)

Reviewer: Maybe group features first and bugs second, instead of intertwining them?

Author: They are sorted by PR number as in previous releases, but I also think grouping them would make it easier to read

> - feature: Change format of content of _validation from defaultdict to dict (`PR#431 <https://github.com/scrapinghub/spidermon/pull/431>`_)  

Reviewer: ```suggestion
- feature: Change the format of the content of _validation from defaultdict to dict (`PR#431 <https://github.com/scrapinghub/spidermon/pull/431>`_)  
```
